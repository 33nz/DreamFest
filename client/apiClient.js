import request from 'superagent'
import L from 'leaflet'

const rootUrl = '/api/v1'

export function getGardens () {
  return request.get(rootUrl + '/gardens')
    .then(res => {
      return res.body.gardens
    })
}

export function getMap () {
  const mymap = L.map('mapid').setView([-36.8666700, 174.7666700], 13)

  L.tileLayer('https://api.mapbox.com/styles/v1/{id}/tiles/{z}/{x}/{y}?access_token={accessToken}', {
    attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a> contributors, <a href="https://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery Â© <a href="https://www.mapbox.com/">Mapbox</a>',
    maxZoom: 18,
    id: 'mapbox/streets-v11',
    tileSize: 512,
    zoomOffset: -1,
    accessToken: 'pk.eyJ1IjoibWljaGFlbC1yIiwiYSI6ImNrZXM1Zm9iaDJiNmYycW1za2dobDZ4d3gifQ.a5mK2DxNqWhlzvoa8Zxb2Q'
  }).addTo(mymap)
}