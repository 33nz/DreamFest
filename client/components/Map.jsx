import React from 'react'

import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet'

function Map (props) {
  console.log('Map.jsx > props.gardens: ', props)
  return (
    <div className='column'>
      <MapContainer
        center={[-36.8666700, 174.7666700]}
        zoom={12}
        scrollWheelZoom={true}>
        <TileLayer
          url='https://api.mapbox.com/styles/v1/mapbox/streets-v11/tiles/{z}/{x}/{y}?access_token=pk.eyJ1IjoibWljaGFlbC1yIiwiYSI6ImNrZXM1Zm9iaDJiNmYycW1za2dobDZ4d3gifQ.a5mK2DxNqWhlzvoa8Zxb2Q'/>
        {props.coordinates?.map(location => {
          return <Marker key={location.lat}
            position={[location.lat, location.lon]}>
            <Popup>
              {props.address}
            </Popup>
          </Marker>
        })}
      </MapContainer>
    </div>
  )
}

export default Map
