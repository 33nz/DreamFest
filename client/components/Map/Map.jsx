import React from 'react'

import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet'

export default function Map ({ addresses, coordinates, userCoordinates }) {
  return (
    <div className='map-container'>
      <MapContainer className='map'
        center={[-36.8666700, 174.7666700]}
        zoom={11}
        scrollWheelZoom={true}>
        <TileLayer
          url='https://api.mapbox.com/styles/v1/mapbox/streets-v11/tiles/{z}/{x}/{y}?access_token=pk.eyJ1IjoibWljaGFlbC1yIiwiYSI6ImNrZXM1Zm9iaDJiNmYycW1za2dobDZ4d3gifQ.a5mK2DxNqWhlzvoa8Zxb2Q'/>
        {coordinates.map((location, i) => {
          return <Marker key={i}
            position={[location.lat, location.lon]}
            draggable={true}
          >
            <Popup>
              {addresses[i]}
            </Popup>
          </Marker>
        })}
        {userCoordinates &&
          <Marker
            position={[userCoordinates.lat, userCoordinates.lon]}
          >
            <Popup>
              Your Location
            </Popup>
          </Marker>
        }
      </MapContainer>
    </div>
  )
}
