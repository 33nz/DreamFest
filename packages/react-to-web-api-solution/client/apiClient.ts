import request from 'superagent'

import { Widget, WidgetData } from '../models/Widget'

const widgetUrl = '/api/v1/widgets/'

export function getWidgets(): Promise<Widget[]> {
  return request.get(widgetUrl).then((res) => res.body)
}

export function addWidget(widget: WidgetData): Promise<Widget[]> {
  return request
    .post(widgetUrl)
    .send(widget)
    .then((res) => res.body)
}
