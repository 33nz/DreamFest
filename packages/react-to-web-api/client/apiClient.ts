// eslint-disable-next-line no-unused-vars
import request from 'superagent'
// eslint-disable-next-line no-unused-vars
const widgetUrl = '/api/v1/widgets/'

export function getWidgets() {
  return request
    .get(widgetUrl)
    .then((response) => {
      return response.body
    })
    .catch((err) => {
      console.error(err.message)
    })
}
