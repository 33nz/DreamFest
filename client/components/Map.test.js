import React from 'react'
import renderer from 'react-test-renderer'

import Map from './Map'
import { getMap } from '../apiClient'

it('renders correctly', () => {
  const tree = renderer
    .create(<Map />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
