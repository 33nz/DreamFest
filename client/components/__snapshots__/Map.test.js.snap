// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="column is-half-tablet"
>
  <div
    id="mapid"
  >
    <p>
      12 Hukanui Crescent
    </p>
  </div>
</div>
`;
